app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False):
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(other: Herbivore):
+        if isinstance(other, Herbivore) and not other.hidden:
+            other.health -= 50
+            if other.health == 0:
```

Comment:
I made this mistace too) What if bitten animal would have health below zero

Reply:
We make hp = 100 by default. How can we get hp < 0, if we only make steps with scale of 50? But it's truly better to take care of unpredicted case.

Reply:
Thanks, that's a  good point. As it is said "If person can make a mistake, he'll definitely do it"